### PR TITLE
IA-1940: Improve "user management" permission

### PR DESCRIPTION
**Pull request by bmonjoie**
Split the user management permission into "admin user manager" and "user management"

Related JIRA tickets : IA-1940

## Self proofreading checklist

- [x] Did I use eslint and black formatters
- [x] Is my code clear enough and well documented
- [x] Are my typescript files well typed
- [x] New translations have been added or updated if new strings have been introduced in the frontend
- [x] My migrations file are included
- [x] Are there enough tests
- [ ] Documentation has been included (for new feature)

## Changes

- Replace the hardcoded permissions with constants
- Implement permission in `/api/profiles/` as well as in `/api/tasks/create/profilesbulkupdate/`
- Slightly change the UI to take into account the new permission

## How to test
Assign the permission "Users (management)" to a user, optionally, assign an org unit to that user to limit its permission to users within that org unit.
Log in with that user and manage users within its health pyramid.

### CODE REVIEW

File: hat/menupermissions/models.py [hunk 1]
```diff
@@ -17,46 +17,75 @@
 `/api/permissions/` endpoint
 """
 
-
 from django.db import models
 from django.utils.translation import gettext_lazy as _
 
-MODIFICATIONS = _("Modifications")
+_ASSIGNMENTS = "iaso_assignments"
```
Comment (madewulf) [inline hat/menupermissions/models.py:25]: Didn't you remove translations here. _() is used to translate the presentation text of a permission, normally

File: hat/menupermissions/models.py [hunk 2]
```diff
@@ -66,57 +95,57 @@ class CustomPermissionSupport(models.Model):
...
-        "iaso_sources",
-        "iaso_data_tasks",
-        "iaso_reports",
+        _FORMS,
+        _SUBMISSIONS,
+        _MAPPINGS,
+        _COMPLETENESS,
+        _ORG_UNITS,
+        _LINKS,
+        _USERS,
```
Comment (madewulf) [inline hat/menupermissions/models.py]: Here, you could fix another existing ticket by including all the permissions for a new account_user

File: hat/menupermissions/models.py (hunk 1 above)
Comment (bmonjoie) [reply hat/menupermissions/models.py:25]: I have moved the translation to the list and used the constants for the keys.

File: hat/menupermissions/models.py (hunk 2 above)
Comment (bmonjoie) [reply hat/menupermissions/models.py]: Could you link the ticket here?

File: iaso/api/profiles.py [hunk 3]
```diff
@@ -15,19 +16,50 @@
...
+    @staticmethod
+    def has_permission_over_user(request, view):
+        pk = view.kwargs.get("pk")
+        if not pk:
+            raise PermissionDenied(f"User with '{permission.USERS_MANAGED}' cannot create users.")
+
+        if pk == request.user.id:
+            raise PermissionDenied(f"User with '{permission.USERS_MANAGED}' cannot edit their own permissions.")
+
+        org_units = OrgUnit.objects.hierarchy(request.user.iaso_profile.org_units.all()).values_list("id", flat=True)
```
Comment (madewulf) [inline iaso/api/profiles.py:53]: I'm questioning this. A user that can modify the permissions of everybody without creating one could be an interesting option too and would maybe match better what we do with org units limitation (basically, you're more powerful if you have no org units assigned than if you have some)

**Comment by madewulf** [comment]
Wondering how all this mixes with the user roles that were recently created and the bulk user update too that we just merged. I have the feeling that we would need to rebase this on main an take these questions into consideration

**Review by beygorghor — APPROVED** [review]

**Review by SimonDus — APPROVED** [review]
I have tested it on my local instance and it seems to work as expected. 
Good job !

**Pull request merged**